pages/checkin.js

```diff
@@ -0,0 +1,204 @@
+import Layout from '../components/Layout'
+import Head from 'next/head'
+import { useEffect, useState } from 'react'
+import cookie from 'js-cookie';
+import React from 'react';
+
+import { useUser } from '../context/userContext'
+import { server } from './_app.js'
+import { render } from 'react-dom';
+
+const fetcher = (url) => fetch(url).then((res) => res.json())
+
+const token = cookie.get("firebaseToken")
+
+class Checkin extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      error: null,
+      success: null,
+      lastScannedID: "N/A",
+      visitsLastWeek: [],
+      lastScannedTime: "N/A"
+    }
+  }
+
+  showError = (errorText, t) => {
+    /* show error banner with error text for 5 seconds, or custom time */
+    this.setState({
+      error: errorText,
+      success: null
+    });
+
+    t = t ? t : 5000;
+    clearTimeout(this.errorTimer);
+    this.errorTimer = setTimeout(() => this.setState({error: null}), t);
+  }
+
+  showSuccess = (msg, t) => {
+    /* show error banner with error text for 5 seconds, or custom time */
+    this.setState({
+      error: null,
+      success: msg
+    });
+
+    t = t ? t : 5000;
+    clearTimeout(this.successTimer);
+    this.successTimer = setTimeout(() => this.setState({success: null}), t);
+  }
+
+  
+  showLastScannedInfo = () => {
+    if (this.state.lastScannedID == "N/A") {
+      return <div className='flex-grow text-center'>Last Scanned ID: {this.state.lastScannedID}</div>
+    }
+    else {
+      return (
+        <>
+        <div className='flex-grow text-center'>Last Scanned ID: {this.state.lastScannedID} at {this.state.lastScannedTime}.</div>
+        </>
+      )
+    }
+  };
+
+  showLastVisitInfo = () => {
```

Comment:
might be slightly easier to read if each date is on a separate line/div? but it also doesn't make much of a difference as long as ID 234 doesn't show up 30 times a week 😄